ttnn/cpp/ttnn/operations/normalization/batch_norm/device/batch_norm_program_factory.cpp

```diff
@@ -0,0 +1,313 @@
+    uint32_t num_cores_y = compute_with_storage_grid_size.y;
+    auto all_device_cores = CoreRange({0, 0}, {num_cores_x - 1, num_cores_y - 1});
+
+    Buffer* a_buffer = a.buffer();
+    Buffer* b_buffer = b.buffer();
+    Buffer* c_buffer = output.buffer();
+    Buffer* d_buffer = d.buffer();
+    Buffer* e_buffer = nullptr;
+    Buffer* f_buffer = nullptr;
+
+    // Number of tiles to store per input CB (double buffer)
+    constexpr uint32_t num_tiles_per_cb = 2;
+    uint32_t b_num_tiles_per_cb = num_tiles_per_cb;
+
+    // Input buffers
+    auto [a_cb, a_cb_handle] = create_cb(
+        tt::CBIndex::c_0, program, all_device_cores, a_single_tile_size, num_tiles_per_cb, a_data_format);  // input
+    auto [b_cb, b_cb_handle] = create_cb(
+        tt::CBIndex::c_1,
+        program,
+        all_device_cores,
+        b_single_tile_size,
+        b_num_tiles_per_cb,
+        b_data_format);  // batch_mean
+    auto [c_cb, c_cb_handle] = create_cb(
+        tt::CBIndex::c_2, program, all_device_cores, c_single_tile_size, num_tiles_per_cb, c_data_format);  // output
+    auto [d_cb, d_cb_handle] = create_cb(
+        tt::CBIndex::c_3,
+        program,
+        all_device_cores,
+        d_single_tile_size,
+        b_num_tiles_per_cb,
+        d_data_format);  // batch_var
+    auto [eps_cb, eps_cb_handle] = create_cb(
+        tt::CBIndex::c_4, program, all_device_cores, d_single_tile_size, b_num_tiles_per_cb, d_data_format);  // eps
+    auto [e_cb, e_cb_handle] = create_cb(
+        tt::CBIndex::c_16, program, all_device_cores, e_single_tile_size, b_num_tiles_per_cb, e_data_format);  // weight
+    auto [f_cb, f_cb_handle] = create_cb(
+        tt::CBIndex::c_18, program, all_device_cores, f_single_tile_size, b_num_tiles_per_cb, f_data_format);  // bias
+
+    // Temporary buffers to store intermediate results
+    auto [den_cb, den_cb_handle] = create_cb(
+        tt::CBIndex::c_5,
+        program,
+        all_device_cores,
+        a_single_tile_size,
+        num_tiles_per_cb,
+        a_data_format);  // to store 1/(sqrt(batch_var + eps))
+    auto [num_cb, num_cb_handle] = create_cb(
+        tt::CBIndex::c_6,
+        program,
+        all_device_cores,
+        a_single_tile_size,
+        num_tiles_per_cb,
+        a_data_format);  // to store input - batch_mean
+    auto [temp_1_cb, temp_1_cb_handle] =
+        create_cb(tt::CBIndex::c_17, program, all_device_cores, a_single_tile_size, num_tiles_per_cb, a_data_format);
+
+    auto a_is_dram = static_cast<uint32_t>(a_buffer->buffer_type() == tt_metal::BufferType::DRAM);
+    auto b_is_dram = static_cast<uint32_t>(b_buffer->buffer_type() == tt_metal::BufferType::DRAM);
+    auto c_is_dram = static_cast<uint32_t>(c_buffer->buffer_type() == tt_metal::BufferType::DRAM);
+    auto d_is_dram = static_cast<uint32_t>(d_buffer->buffer_type() == tt_metal::BufferType::DRAM);
+    bool e_is_dram = false;
+    bool f_is_dram = false;
```

Comment:
```suggestion
    const auto e_is_dram = weight_has_value and e->buffer_type() == tt_metal::BufferType::DRAM;
    const auto f_is_dram = bias_has_value and f->buffer_type() == tt_metal::BufferType::DRAM;
```
and delete the declarations for `e_buffer` and `f_buffer` above, and the two respective conditional blocks below.